### DISCUSSION TITLE
Abort a zipping process.

### DISCUSSION THREAD

**Discussion by mdeangelo-CT** [icsharpcode/SharpZipLib#598] [Q&A]
### Expected behavior
Create a method to abort zipping in FastZip. As seen in issue #373, we see that it is unsafe to abort a running thread. Could we implement a method that could safely abort the zipping process?

### Version of SharpZipLib
1.3.1

### Obtained from (only keep the relevant lines)
- Package installed using NuGet

**Comment by piksel** [comment]
You can use the `ZipFile.CompletedFileHandler` event and cancel the compression there if you want, something like:

```cs
var fastZip = new FastZip(new FastZipEvents() {
    CompletedFile = (_, args) => { 
        args.ContinueRunning = !aborted;
    }
});

fastZip.CreateZip(zipFileName, sourceFolderPath, true, "");
```

Note that this is single threaded, so if you want to listen for some kind of external event you probably have to do threading and sync/lock

**Reply by mdeangelo-CT** [reply]
Hello,

I have implemented this, and it still doesn't work for me. Perhaps you could point me in the correct direction if I am doing something wrong.

I am controlling the aborted variable on a different thread. When an event gets called, abort is switched to true. I verified through my debugger that args.ContinueRunning is successfully changed to false, but that does not stop the process. Is this what you were referring to when you mentioned threading and locking, and if so, what am I missing?

I really appreciate the help,
DeAngelo
